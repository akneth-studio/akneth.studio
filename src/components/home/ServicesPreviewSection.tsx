import CTAButton from "../CTAButton";

export default function ServicesPreviewSection() {
  return (
    <section className="services-preview bg-light py-5 px-4">
      <div className="container">
        <h2>Usługi</h2>
        <ul className="list-unstyled">
          <li><strong>Szablony Google</strong> - projektowanie funkcjonalnych arkuszy dla firm i osób prywatnych.</li>
          <li><strong>Automatyzacje</strong> - usprawnianie codziennej pracy dzięki prostym, praktycznym rozwiązaniom.</li>
        </ul>
        <CTAButton
          type='button'
          text="Zobacz pełną ofertę"
          variant="outline-primary"
          to="/services"
        />
      </div>
    </section>
  );
}
