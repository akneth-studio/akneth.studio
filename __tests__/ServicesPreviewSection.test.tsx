import React from 'react';
import { render, screen } from '@testing-library/react';
import ServicesPreviewSection from '../src/components/home/ServicesPreviewSection';
import '@testing-library/jest-dom';

describe('ServicesPreviewSection component', () => {
  it('renders heading and service previews', () => {
    render(<ServicesPreviewSection />);
    expect(screen.getByRole('heading', { name: /usÅ‚ugi/i })).toBeInTheDocument();
    // Add more specific checks if the component renders specific service items or text
  });

  it('matches snapshot', () => {
    const { container } = render(<ServicesPreviewSection />);
    expect(container).toMatchSnapshot();
  });
});
