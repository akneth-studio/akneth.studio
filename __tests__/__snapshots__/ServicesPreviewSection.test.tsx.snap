// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ServicesPreviewSection component matches snapshot 1`] = `
<div>
  <section
    class="services-preview bg-light py-5 px-4"
  >
    <div
      class="container"
    >
      <h2>
        Usługi
      </h2>
      <ul
        class="list-unstyled"
      >
        <li>
          <strong>
            Szablony Google
          </strong>
           - projektowanie funkcjonalnych arkuszy dla firm i osób prywatnych.
        </li>
        <li>
          <strong>
            Automatyzacje
          </strong>
           - usprawnianie codziennej pracy dzięki prostym, praktycznym rozwiązaniom.
        </li>
      </ul>
      <button
        aria-label="Zobacz pełną ofertę"
        class="btn btn-outline-primary  btn btn-primary"
        data-testid="cta-button"
        role="button"
        tabindex="0"
        type="button"
      >
        Zobacz pełną ofertę
      </button>
    </div>
  </section>
</div>
`;
